src/main/java/de/cas/mse/exercise/SumDigPower.java

```diff
@@ -5,37 +5,47 @@
 
 public class SumDigPower {
 
-	public List<Long> sumDigPow(long a, long b) {
+    List<Long> eurekaList = new ArrayList<Long>();
 
-		List<Long> result = new ArrayList<Long>();
+	public List<Long> sumDigPow(long lowerLimit, long upperLimit) {
 
-		for (long i = a; i < b; i++) {
+		for (long currentNumber = lowerLimit; currentNumber < upperLimit; currentNumber++) {
 
-			List<Long> longs = new ArrayList<Long>();
+			List<Long> currentNumberSplitted = new ArrayList<Long>();
 
-			String temp = Long.toString(i);
+			String numberAsString = Long.toString(currentNumber);
 
-			// Split values
-			for (int j = 0; j < temp.length(); j++) {
-				longs.add(Long.valueOf(temp.substring(j, j + 1)));
-			}
+			splitValues(currentNumberSplitted, numberAsString);
 
-			// Create sum
-			long sum = 0;
-			for (int j = 1; j <= longs.size(); j++) {
-				sum += Math.pow(longs.get(j - 1), j);
-			}
+			long sum = createSum(currentNumberSplitted);
 
-			// Test if sum is equal
-			if (sum == i) {
-				result.add(sum);
-			}
+			addToEurekaListIfEqual(currentNumber, sum);
 
 		}
 
-        System.out.println(result);
-		return result;
+        System.out.println(eurekaList);
+		return eurekaList;
 
 	}
 
+    private void addToEurekaListIfEqual(long currentNumber, long sum) {
```

Comment:
Maybe an own method for this is too much. Then the global variable can be removed too